File: src/stores/product.js
import { defineStore } from 'pinia'
import { ref } from 'vue'

export const useProductStore = defineStore('product', () => {
  const products = ref([])
  const currentProduct = ref(null)

  // 获取产品列表
  const fetchProducts = async () => {
    // 模拟获取产品数据
    products.value = [
      { id: 1, name: '淮南豆腐文化文创', price: 99, description: '精美的豆腐文化创意产品' },
      { id: 2, name: '八公山传说书籍', price: 68, description: '八公山历史文化书籍' }
    ]
  }

  return {
    products,
    currentProduct,
    fetchProducts
  }
})

File: hyyz-backend/hyyz-backend/src/main/java/com/hyyz/entity/Product.java
package com.hyyz.entity;

import lombok.Data;
import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    @Enumerated(EnumType.STRING)
    private ProductCategory category;
    
    private String description;
    
    @Column(columnDefinition = "TEXT")
    private String content;
    
    @ElementCollection
    private List<String> images;
    
    private Double price;
    private Double originalPrice;
    private Integer stock = 0;
    private Integer sales = 0;
    
    @ElementCollection
    private List<String> tags;
    
    @Enumerated(EnumType.STRING)
    private ContentStatus status = ContentStatus.PUBLISHED;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id")
    private User author;
    
    @Column(updatable = false)
    private LocalDateTime createTime;
    
    private LocalDateTime updateTime;
    
    @PrePersist
    protected void onCreate() {
        createTime = LocalDateTime.now();
        updateTime = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updateTime = LocalDateTime.now();
    }
    
    public enum ProductCategory {
        FOOD, CRAFT, STATIONERY, CLOTHING, HOME
    }
}